node_modules/oae-principals/lib/internal/dao.js

```diff
@@ -59,8 +61,10 @@ var createUser = module.exports.createUser = function(user, callback) {
     };
     queries.push(Cassandra.constructUpsertCQL('Principals', 'principalId', user.id, values));
 
-    // Map the email address to the user id
-    queries.push(Cassandra.constructUpsertCQL('PrincipalsByEmail', 'email', user.email, {'principalId': user.id}));
+    // Only map the email address to the user if it's verified
+    if (user.emailVerified && user.email) {
+        queries.push({'query': 'INSERT INTO "PrincipalsByEmail" ("email", "principalId") VALUES (?, ?)', 'parameters': [user.email, user.id]});
+    }
```

Comment:
I would prefer if `PrincipalsByEmail` was just a legit "secondary index" on the Principals `email` field, and not have some weird cases where you can have an email exist in Principals that doesn't exist in PrincipalsByEmail. I think if we make `user.email` existing imply verification, then this would become the case.